packages/analytics-js/src/components/utilities/json.ts

```diff
@@ -18,11 +22,12 @@ const getCircularReplacer = (excludeNull?: boolean): ((key: string, value: any)
     // `this` is the object that value is contained in, i.e., its direct parent.
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore-next-line
-    while (ancestors.length > 0 && ancestors.at(-1) !== this) {
+    while (ancestors.length > 0 && ancestors[ancestors.length - 1] !== this) {
       ancestors.pop();
     }
 
     if (ancestors.includes(value)) {
+      logger?.warn(`Circular Reference detected and dropped fro property: ${key}`);
```

Comment:
logger?.warn(`Circular Reference detected and dropped from property: ${key}`);

Reply:
done